pkg/resourcegroup/controllers/resourcegroup/resourcegroup_controller.go

```diff
@@ -346,23 +348,112 @@ func (r *reconciler) computeStatus(
 	return statuses
 }
 
-// ActuationStatusToLegacy contains the logic/rules to convert from the actuation statuses
-// to the legacy status field. If conversion is not needed, the original status field is returned
-// instead.
-func ActuationStatusToLegacy(s v1alpha1.ResourceStatus) v1alpha1.Status {
-	if s.Status == v1alpha1.NotFound {
-		return v1alpha1.NotFound
+// UpdateReconcileStatusToReflectKstatus uses the current Strategy and Actuation
+// status from the applier and the newly computed kstatus to compute the
+// Reconcile status. Returns an error if one of the inputs is invalid.
+func UpdateReconcileStatusToReflectKstatus(status v1alpha1.ResourceStatus) (v1alpha1.Reconcile, error) {
+	switch status.Strategy {
+	case v1alpha1.Apply:
+		switch status.Actuation {
+		case v1alpha1.ActuationSucceeded:
+			switch status.Status {
+			case v1alpha1.Current:
+				return v1alpha1.ReconcileSucceeded, nil // Strategy=Apply + Actuation=Succeeded + kstatus=Current => Reconcile=Succeeded
+			case v1alpha1.InProgress:
+				return v1alpha1.ReconcilePending, nil // Strategy=Apply + Actuation=Succeeded + kstatus=InProgress => Reconcile=Pending
+			case v1alpha1.Failed:
+				return v1alpha1.ReconcileFailed, nil // Strategy=Apply + Actuation=Succeeded + kstatus=Failed => Reconcile=Failed
+			case v1alpha1.Terminating:
+				return v1alpha1.ReconcileFailed, nil // Strategy=Apply + Actuation=Succeeded + kstatus=Terminating => Reconcile=Failed // TODO: Should this be Pending?
```

Comment:
TODO: Should this be Pending, instead of Failed?

If we assume Failed is a terminal state that needs user interaction, then this isn't really failed. We could mark it as Pending instead, because the applier needs to re-actuate in order to re-apply the deleted object.

However, if the applier succeeded syncing, it might not be retrying. So it might not re-actuate for another hour.

Plus, if something deleted the object after config sync applied it, there's probably actually a controller fight, or some manual change by a user, which might indicate it needs user interaction to fix anyway.

@janetkuo What do you think?